### ISSUE TITLE
debug no longer works with browserify

### ISSUE THREAD

**Issue by wraithgar** [visionmedia/debug#606]
Any version of debug that I use past version 3.1.0 gives this error when trying to browserify code that includes this module

```
Cannot find module './common' from '[redacted]/node_modules/debug/dist'
      at FSReqWrap.oncomplete (fs.js:154:21)
```

There was a mention in #603 about a browserify incompatibility, but it doesn't appear to be the same issue.

**Comment by Qix-** [comment]
Hi - which version **exactly** is being pulled down?

**Comment by Qix-** [comment]
Interesting, I'll take a look - give me a moment.

**Comment by gztomas** [comment]
@Qix- maybe it is that we are using `martifactory` registry as a caching proxy in the middle which is not aware yet of that deprecation. I will look into that.

**Comment by kolegm** [comment]
the same issue for me with debug@4.0.0
`Reason: Error: Cannot find module './common' from 'node_modules/debug/dist'`

**Comment by Qix-** [comment]
Does the code on unpkg match the code in the `node_modules/debug/dist/debug.js` folder?

https://unpkg.com/debug@4.0.0/dist/debug.js

If so, then there's no reason this shouldn't work.

<img width="1006" alt="image" src="https://user-images.githubusercontent.com/885648/45383329-1d08c200-b5c8-11e8-8a96-1db2aae7c24e.png">
<img width="1006" alt="image" src="https://user-images.githubusercontent.com/885648/45383336-209c4900-b5c8-11e8-93cc-45f783972fd5.png">

**Comment by wraithgar** [comment]
Deep requiring into a specific file in the package from an unpkg url is not exactly a "works in the browser" test that feels like it was made in good faith.  Here's a barebones "try debug with browserify" that breaks (assumes you have npm@6 so that npx is a command you can run):

```sh
$ npm init -y
$ npm install debug@latest browserify@latest
$ echo "var log = require('debug')('example')" >example.js
$ npx browserify example.js
```
```
Error: Cannot find module './common' from '/usr/local/src/gar/projects/example/node_modules/debug/dist'
    at /usr/local/src/gar/projects/example/node_modules/browser-resolve/node_modules/resolve/lib/async.js:55:21
    at load (/usr/local/src/gar/projects/example/node_modules/browser-resolve/node_modules/resolve/lib/async.js:69:43)
    at onex (/usr/local/src/gar/projects/example/node_modules/browser-resolve/node_modules/resolve/lib/async.js:92:31)
    at /usr/local/src/gar/projects/example/node_modules/browser-resolve/node_modules/resolve/lib/async.js:22:47
    at FSReqWrap.oncomplete (fs.js:154:21)
```

**Comment by Qix-** [comment]
> Deep requiring into a specific file in the package from an unpkg url is not exactly a "works in the browser" test that feels like it was made in good faith. 

Feel free to submit a PR that automates this testing. The release wasn't made in "bad faith", if that's what you're suggesting. I'm one person managing a package that is downloaded several million times per day - all I can do is help fix problems where they arise.

The alternative is to not release ever again, which solves nothing.

---

Thank you for the reproduction steps - I can confirm they reproduce.

**Comment by Qix-** [comment]
Some input on #607 would be appreciated. You can test this version by running `npm i 'visionmedia/debug#fix-browserify'`.

**Comment by gztomas** [comment]
Could this be included in a `3.2.5` release? We are still getting `3.2.4` when using `yarn`

**Comment by Qix-** [comment]
@gztomas sure.

**Comment by Qix-** [comment]
Released as `4.0.1` and backported to `3.2.5`.

Even though this issue is closed let me know _here_ if that doesn't solve it.

**Comment by gztomas** [comment]
Working for us! Thanks for the awesome support!

**Comment by Qix-** [comment]
No problem, thank you for following up @gztomas

**Comment by ThomasGHenry** [comment]
Back in business, @Qix- ! Thanks for your hard work! I do not envy your position these past several hours. What a nightmare, but well met!

**Comment by Qix-** [comment]
Thank you @ThomasGHenry, sorry for the trouble :)

**Comment by nealyip** [comment]
Hi, I found this fix broke Angular/Webpack.
I am a user of aws-appsync. Since a recent release, the debug package is added to their dependencies.
However, IE11 no longer worked as Angular/Webpack utilized the browser property from debug's package.json when bundling. 

It worked if I edited the file node_modules/aws-appsync/debug/package.json and turned browser back to "./dist/debug.js" as webpack used that file instead of src/browser.js for a browser.
